dig.go

```diff
@@ -265,10 +256,50 @@ var _graphTmpl = template.Must(
 // Visualize parses the graph in Container c into DOT format and writes it to
 // io.Writer w.
 func Visualize(c *Container, w io.Writer, opts ...VisualizeOption) error {
-	if err := _graphTmpl.Execute(w, c.dg); err != nil {
-		return fmt.Errorf("error executing template: %v", err)
+	return _graphTmpl.Execute(w, c.createGraph())
+}
+
+// VisualizeError parses the container and errors included in err into a DOT
+// format graph and writes it to io.Writer w.
+func VisualizeError(err error, w io.Writer, opts ...VisualizeOption) error {
+	ep, ok := err.(errEntryPoint)
+	if !ok {
+		return fmt.Errorf("no graph included in error: %v", err)
```

Comment:
Nice. I think we may be able to drop this if we make `VisualizeError` an
option as discussed offline earlier. So,

    Visualize(c, w)  // visualizes the container
    Visualize(c, w, VisualizeError(err))  // visualizes the error in the container visualization
    
That way, the container is always available.

Thoughts? @amckinney @leyao

Reply:
I was actually just making the same comment :) - let's go with the option.

Reply:
But as @abhinav suggests, let's handle this in a follow-up.

Reply:
Great suggestion, that way we can avoid passing around the container in the errors. I'll work on this next!